### PR TITLE
[CDAP-6501] UI Custom Actions Support

### PR DESCRIPTION
**Pull request by elfenheart**
screen shot 2016-07-20 at 4 21 00 pm

### CODE REVIEW

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 1]
```diff
@@ -22,10 +22,16 @@ angular.module(PKG.name + '.commons')
     var numberFilter = $filter('number');
 
     var endpoints = [];
-    var sourceSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).source);
-    var sinkSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).sink);
-    var transformSourceSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).transformSource);
-    var transformSinkSettings = angular.copy(DAGPlusPlusFactory.getSettings(false).transformSink);
+
+    var sourceOrigin = angular.copy(DAGPlusPlusFactory.getSettings().sourceOrigin),
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js]: We should probably do something like 

```
angular.copy(DAGPlusPlusFactory.getSettings())
```

and then use objects from it. `angular.copy` is a costly operation.

File: cdap-ui/app/directives/dag-plus/my-dag-factory.js [hunk 2]
```diff
@@ -56,39 +43,51 @@ angular.module(PKG.name + '.commons')
...
-
-      settings.transformSource = angular.copy(settings.source);
-      settings.transformSink = angular.copy(settings.sink);
-      settings.transformSource.anchor = [ 0.5, 1, 1, 0, 26, -43, 'transformAnchor'];
-      settings.transformSink.anchor = [ 0.5, 1, -1, 0, -26, -43, 'transformAnchor'];
+    function getSettings() {
+      var settings = {
+        default: defaultSettings,
+        commonSettings: angular.extend(commonSettings, connectorOverlays),
+        sourceOrigin: angular.copy(originSettings),
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/dag-plus/my-dag-factory.js:67]: If we are doing `angular.copy` while performing `getSettings` do we need to do it here? We should try doing it in one place once..

File: cdap-ui/app/directives/dag-plus/my-dag.less [hunk 3]
```diff
@@ -136,6 +137,109 @@ my-dag-plus {
         }
       }
 
+      &.action {
+        @brown: #988470;
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/dag-plus/my-dag.less]: May be at the top of the file? Took a little while to see that this is a declaration.

**Comment by ajainarayanan** [comment]
Minor comments. Otherwise worked fabulously! 👍 🎱  Awesome work @elfenheart!

File: cdap-ui/app/directives/dag-plus/my-dag-factory.js (hunk 2 above)
Comment (elfenheart) [inline cdap-ui/app/directives/dag-plus/my-dag-factory.js:67]: we should do it here, since I am making modification to the content

**Comment by elfenheart** [comment]
demoed to @sreevatsanraman

**Pull request merged**